### PR TITLE
feat: emit end to end latency metric from rest-hook Lambda

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Emit end to end latency metric from rest-hook Lambda

Screen Shot 2022-02-18 at 2 23 12 PM

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/subscriptions/restHookLambda/restHook.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import axios from 'axios';
 import { makeLogger } from 'fhir-works-on-aws-interface';
 import { SQSEvent } from 'aws-lambda';
 import { SubscriptionNotification } from 'fhir-works-on-aws-search-es';
+import { metricScope, Unit } from 'aws-embedded-metrics';
+import moment from 'moment';
```
Comment (carvantes) [inline src/subscriptions/restHookLambda/restHook.ts]: I'd suggest not using `moment`

The difference in milliseconds is easy to calculate using the available Date functions
```
Date.now() - (new Date('2022-02-17T20:33:57.021Z')).getTime()
```

Also, moment is sort of deprecated nowadays: https://github.com/moment/moment/#project-status

File: src/subscriptions/restHookLambda/restHook.ts [hunk 2]
```diff
@@ -19,15 +21,29 @@ const mergeRequestHeaders = (allowListHeader: string[], channelHeader: string[])
...
     allowListHeader.forEach((header) => mergeHeader(header));
     channelHeader.forEach((header) => mergeHeader(header));
     return mergedHeader;
 };
 
+/**
+ * Push latency metric to CloudWatch
+ * @param messages
+ */
+const pushLatencyMetric = metricScope((metrics) => async (messages: SubscriptionNotification[]): Promise<void> => {
```
Comment (carvantes) [inline src/subscriptions/restHookLambda/restHook.ts]: nit: rename to something like `recordLatencyMetric` or ``logLatencyMetric``. I think that "push" sounds misleading

**Review by carvantes — CHANGES_REQUESTED** [review]

File: src/subscriptions/restHookLambda/restHook.ts (hunk 1 above)
Comment (Bingjiling) [reply src/subscriptions/restHookLambda/restHook.ts]: Wow, I didn't know moment is getting deprecated. Will update to Date.

File: src/subscriptions/restHookLambda/restHook.ts [hunk 3]
```diff
@@ -19,15 +20,29 @@ const mergeRequestHeaders = (allowListHeader: string[], channelHeader: string[])
...
+ * Push latency metric to CloudWatch
+ * @param messages
+ */
+const logLatencyMetric = metricScope((metrics) => async (messages: SubscriptionNotification[]): Promise<void> => {
+    const currentTime = new Date().getTime();
+    messages.forEach((message: SubscriptionNotification) => {
+        metrics.putMetric(
+            'SubscriptionEndToEndLatency',
+            (currentTime - new Date(message.matchedResource.lastUpdated).getTime()) / 1000,
+            Unit.Seconds,
```
Comment (carvantes) [inline src/subscriptions/restHookLambda/restHook.ts]: It's almost always better to emit metrics in the smallest unit available. In this case Milliseconds.

When storing milliseconds you can always divide by 1000 on the graphs if you want to see seconds. But storing the metric as seconds destroys the milliseconds information.

**Review by carvantes — CHANGES_REQUESTED** [review]

**Review by carvantes — APPROVED** [review]

**Pull request merged**